File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    const n = prices.length;
    let ans = 0;
    for( let i = 1 ; i < n ; i++){
        if( prices[i] > prices[i-1])
            ans += (prices[i] - prices[i-1]);
    }
    return ans;
};

File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int dp[n+1][2];
        memset(dp, 0, sizeof(dp));
        for( int i = n-1 ; i >= 0 ; i--){
            for( int j = 1 ; j >= 0 ; j--){
                if( j ) dp[i][j] = max(-prices[i] + dp[i+1][0], dp[i+1][1]);
                else  dp[i][j] = max(prices[i]+ dp[i+1][1] , dp[i+1][0]);
            }
        }
        return dp[0][1];
    }
};